apps/omg_watcher_info/lib/omg_watcher_info/api/transaction.ex

```diff
@@ -107,28 +108,100 @@ defmodule OMG.WatcherInfo.API.Transaction do
     end
   end
 
+  @doc """
+  Converts parameter keyword list to a map before passing it to multi-clause "handle_merge/`1"
+  """
+  @spec merge(Keyword.t()) :: create_t()
+  def merge(parameters) do
+    parameters |> Map.new() |> handle_merge()
+  end
+
+  @spec handle_merge(map()) :: create_t()
+  defp handle_merge(%{address: address, currency: currency}) do
+    merge_inputs =
+      address
+      |> DB.TxOutput.get_sorted_grouped_utxos(:asc)
+      |> Map.get(currency, [])
+
+    case merge_inputs do
+      [] ->
+        {:error, :no_inputs_found}
+
+      [_single_input] ->
+        {:error, :single_input}
+
+      inputs ->
+        {:ok, TransactionCreator.generate_merge_transactions(inputs)}
+    end
+  end
+
+  defp handle_merge(%{utxo_positions: utxo_positions}) do
+    with {:ok, inputs} <- get_merge_inputs(utxo_positions),
+         :ok <- no_duplicates(inputs),
+         :ok <- single_owner(inputs),
+         :ok <- single_currency(inputs) do
+      {:ok, TransactionCreator.generate_merge_transactions(inputs)}
```

Comment:
Maybe we can unify this by using the `respond` func to match other functions below?